objectivec/src/ort_session.mm

```diff
@@ -213,6 +213,24 @@ - (nullable instancetype)initWithError:(NSError**)error {
   ORT_OBJC_API_IMPL_CATCH_RETURNING_NULLABLE(error)
 }
 
+- (BOOL)appendExecutionProvider:(NSString*)providerName
+                providerOptions:(NSDictionary*)providerOptions
+                          error:(NSError**)error {
+  std::unordered_map<std::string, std::string> options;
+  NSArray* keys = [providerOptions allKeys];
+
+  for (NSString* key in keys) {
+    NSString* value = [providerOptions objectForKey:key];
+    options.emplace(key.UTF8String, value.UTF8String);
+  }
+
+  try {
```

Comment:
nit: can we put the try/catch around the whole implementation in this function?

Reply:
Sure.